clients/drcachesim/tools/record_filter_launcher.cpp

```diff
@@ -59,6 +61,48 @@ static droption_t<unsigned int> op_verbose(DROPTION_SCOPE_ALL, "verbose", 0, 0,
                                            "Verbosity level",
                                            "Verbosity level for notifications.");
 
+static droption_t<uint64_t>
+    op_stop_timestamp_us(DROPTION_SCOPE_ALL, "stop_timestamp_us", 0, 0,
+                         std::numeric_limits<uint64_t>::max(),
+                         "Timestamp (in us) in the trace when to stop filtering.",
+                         "Record filtering will be disabled (everything will be output) "
+                         "when the tool sees a TRACE_MARKER_TYPE_TIMESTAMP marker with "
+                         "timestamp greater than the specified value.");
+
+static droption_t<int> op_cache_filter_size(
+    DROPTION_SCOPE_FRONTEND, "cache_filter_size", 0,
+    "[Required] Enable data cache filter with given size (in bytes).",
+    "Enable data cache filter with given size (in bytes), with 64 byte "
+    "line size and a direct mapped LRU cache.");
+
+static droption_t<std::string> op_remove_trace_types(
+    DROPTION_SCOPE_FRONTEND, "remove_trace_types", "",
+    "[Required] Comma-separated integers for trace types to remove.",
+    "Comma-separated integers for trace types to remove. "
+    "See trace_type_t for the list of trace entry types.");
+
+static droption_t<std::string> op_remove_marker_types(
+    DROPTION_SCOPE_FRONTEND, "remove_marker_types", "",
+    "[Required] Comma-separated integers for marker types to remove.",
+    "Comma-separated integers for marker types to remove. "
+    "See trace_marker_type_t for the list of marker types.");
+
+template <typename T>
+std::vector<T>
+parse_string(std::string s, char sep = ',')
+{
+    size_t pos;
+    if (s.empty())
+        return {};
+    std::vector<T> vec;
+    do {
+        pos = s.find(sep);
+        vec.push_back(static_cast<T>(std::stoi(s.substr(0, pos))));
+        s.erase(0, pos + 1);
```

Comment:
Do we need to erase and destroy the string?  We can pass the current position to the other funcs and let the string param be a const which seems more the normal case for string params.

Reply:
Using the find(sep, pos) overload now so that we don't have to modify the string.